pytorch_lightning/plugins/precision/apex_amp.py

```diff
@@ -47,13 +48,17 @@ def main_params(self, optimizer: Optimizer) -> _PARAMETERS:
 
     def dispatch(self, trainer: "pl.Trainer") -> None:
         if not self._connected:
-            accelerator = trainer.accelerator
-            _, accelerator.optimizers = amp.initialize(
-                trainer.lightning_module, accelerator.optimizers, opt_level=self.amp_level
+            strategy = trainer.training_type_plugin
+            _, strategy.optimizers = amp.initialize(
+                trainer.lightning_module, strategy.optimizers, opt_level=self.amp_level
             )
             self._connected = True
         return super().dispatch(trainer)
 
+    @property
```

Comment:
Let's move properties to the top, after `__init__` and before the methods? Same would apply for other plugins

Reply:
Make sense, will do for all files. Will also reorg the function orders in all plugins